core/server/master/src/main/java/alluxio/master/file/meta/InodeTree.java

```diff
@@ -1008,7 +1010,23 @@ public void syncPersistDirectory(InodeDirectory dir, JournalContext journalConte
           MkdirsOptions mkdirsOptions =
               MkdirsOptions.defaults().setCreateParent(false).setOwner(dir.getOwner())
                   .setGroup(dir.getGroup()).setMode(new Mode(dir.getMode()));
-          ufs.mkdirs(ufsUri, mkdirsOptions);
+          if (!ufs.mkdirs(ufsUri, mkdirsOptions)) {
+            // Directory already exists. Try loading the status from ufs.
```

Comment:
I think returning false just means the directory couldn't be created, not that it already exists. It could return false due to a permission issue or because the UFS temporarily unavailable

Reply:
I think you are right - it's not restricted to directory exist case. 
In general we want to try creating the directory first, and if it does not work, then it is likely already exist in UFS so we will do a  `getStatus` to grab the metadata. if that still fails then we will report error.

Reply:
I updated the comments to be more accurate.